File: agents/typescript/boilerplate/src/create-workflow.ts
import {AsteroidAgents} from "asteroid-odyssey";
import {CreateWorkflowRequest} from "asteroid-odyssey/dist/generated/agents";

const apiKey: string = process.env.ASTEROID_API_KEY ?? "";
const asteroid = new AsteroidAgents(apiKey);

/**
 * Ceres is our browser running agent that can execute tasks for you. This is the only one at the moment,
 * but there's more coming ASAP.
 */
const agentName: string = "ceres";

const request: CreateWorkflowRequest = {
  name: "Your workflow name",
  fields: {
    "workflow_name": "Your workflow name",
    "start_url": "https://www.google.com",
  },
  prompts: ["Prompt for the agent"],
  provider: CreateWorkflowRequest.provider.ANTHROPIC
}

async function createWorkflow() {
  try {
    const workflowId = await asteroid.createWorkflow(agentName, request);
    console.log('Workflow created with ID:', workflowId);
  } catch (error) {
    console.error('Error executing the workflow:', error);
  }
}

createWorkflow();

File: agents/python/boilerplate/create_workflow.py
import os

from asteroid_odyssey.api.generated.agents.asteroid_agents_api_client.models import CreateWorkflowRequest, CreateWorkflowRequestFields, \
    CreateWorkflowRequestProvider
from asteroid_odyssey import Odyssey


api_key = os.getenv('ASTEROID_API_KEY', '')
asteroid = Odyssey(api_key)

agent_name = "ceres"

request = CreateWorkflowRequest(
    name="Reddit Summariser- PY",
    fields=CreateWorkflowRequestFields.from_dict({
        "workflow_name": "Reddit Summariser- PY",
        "start_url": "https://www.google.com",
    }),
    prompts=["Go to reddit, find a post and summarize it"],
    provider=CreateWorkflowRequestProvider.ANTHROPIC
)

def create_workflow():
    try:
        workflow_id = asteroid.create_workflow(agent_name, request)
        print('Workflow created with ID:', workflow_id)
    except Exception as error:
        print('Error executing the workflow:', error)

if __name__ == '__main__':
    create_workflow()
